src/site/content/en/blog/bluetooth/index.md

```diff
@@ -0,0 +1,549 @@
+`service.getCharacteristic`.
+
+Note that you can also add a `characteristicvaluechanged` event listener on a
+characteristic to handle reading its value. Check out the [Read Characteristic
+Value Changed Sample] to see how to optionally handle upcoming GATT
+notifications as well.
+
+```js
+...
+.then(characteristic => {
+  // Set up event listener for when characteristic value changes.
+  characteristic.addEventListener('characteristicvaluechanged',
+                                  handleBatteryLevelChanged);
+  // Reading Battery Level...
+  return characteristic.readValue();
+})
+.catch(error => { console.log(error); });
+
+function handleBatteryLevelChanged(event) {
+  const batteryLevel = event.target.value.getUint8(0);
+  console.log('Battery percentage is ' + batteryLevel);
+}
+```
+
+### Write to a Bluetooth Characteristic
+
+Writing to a Bluetooth GATT Characteristic is as easy as reading it. This time,
+let's use the Heart Rate Control Point to reset the value of the Energy Expended
+field to 0 on a heart rate monitor device.
+
+I promise there is no magic here. It's all explained in the [Heart Rate Control
+Point Characteristic page].
+
+```js
+navigator.bluetooth.requestDevice({ filters: [{ services: ['heart_rate'] }] })
+.then(device => device.gatt.connect())
+.then(server => server.getPrimaryService('heart_rate'))
+.then(service => service.getCharacteristic('heart_rate_control_point'))
+.then(characteristic => {
+  // Writing 1 is the signal to reset energy expended.
+  const resetEnergyExpended = Uint8Array.of(1);
+  return characteristic.writeValue(resetEnergyExpended);
+})
+.then(_ => {
+  console.log('Energy expended has been reset.');
+})
+.catch(error => { console.log(error); });
+```
+
+### Receive GATT Notifications
+
+Now, let's see how to be notified when the [Heart Rate Measurement]
+characteristic changes on the device:
+
+```js
+navigator.bluetooth.requestDevice({ filters: [{ services: ['heart_rate'] }] })
+.then(device => device.gatt.connect())
+.then(server => server.getPrimaryService('heart_rate'))
+.then(service => service.getCharacteristic('heart_rate_measurement'))
+.then(characteristic => characteristic.startNotifications())
+.then(characteristic => {
+  characteristic.addEventListener('characteristicvaluechanged',
+                                  handleCharacteristicValueChanged);
+  console.log('Notifications have been started.');
+})
+.catch(error => { console.log(error); });
+
+function handleCharacteristicValueChanged(event) {
+  const value = event.target.value;
+  console.log('Received ' + value);
+  // TODO: Parse Heart Rate Measurement value.
+  // See https://github.com/WebBluetoothCG/demos/blob/gh-pages/heart-rate-sensor/heartRateSensor.js
+}
+```
+
+The [Notifications Sample] will show you to how to stop notifications with
+`stopNotifications()` and properly remove the added `characteristicvaluechanged`
+event listener.
+
+### Get disconnected from a Bluetooth Device
```

Comment:
```suggestion
### Disconnect from a Bluetooth Device
```
You can ignore this suggestion if there's a technical difference between your original phrasing and what I'm proposing here

Reply:
Done